### PR TITLE
Keep the `training_step*` output type

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

Fixes https://github.com/PyTorchLightning/pytorch-lightning/issues/9968

TODO:
- Docs?
- Test failures?

### Does your PR introduce any breaking changes? If yes, please list them.

None

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

### CODE REVIEW

File: tests/loops/test_flow_warnings.py [hunk 1]
```diff
@@ -1,45 +0,0 @@
...
-from tests.helpers.boring_model import BoringModel
-
-
-class TestModel(BoringModel):
-    def training_step(self, batch, batch_idx):
-        acc = self.step(batch[0])
-        return acc
-
-
-def test_no_depre_without_epoch_end(tmpdir):
```
Comment (carmocca) [inline tests/loops/test_flow_warnings.py:26]: This was a check for a warning in https://github.com/PyTorchLightning/pytorch-lightning/pull/3849 but it doesn't exist anymore

File: pytorch_lightning/loops/utilities.py [hunk 2]
```diff
@@ -221,3 +224,31 @@ def _reset_progress(loop: Loop) -> None:
...
             _reset_progress(v)
+
+
+# TODO: remove in v1.8
+def _v1_8_output_format(fx: Callable) -> bool:
+    parameters = inspect.signature(fx).parameters
+    return "new_format" in parameters and parameters["new_format"].default is True
+
+
+# FIXME: remove in v1.8
```
Comment (carmocca) [inline pytorch_lightning/loops/utilities.py:235]:
```suggestion
# TODO: remove in v1.8
```

File: pytorch_lightning/loops/optimization/closure.py [hunk 3]
```diff
@@ -13,7 +13,7 @@
 # limitations under the License.
```
Comment (carmocca) [inline pytorch_lightning/loops/optimization/closure.py:13]: Blocked by:

```python
def test_kk():
    import numpy
    batch_output = [torch.tensor([0.1])]
    # FutureWarning: The input object of type 'Tensor' is an array-like implementing one of the corresponding protocols
    # (`__array__`, `__array_interface__` or `__array_struct__`); but not a sequence (or 0-D). In the future, this
    # object will be coerced as if it was first converted using `np.array(obj)`. To retain the old behaviour, you have
    # to either modify the type 'Tensor', or assign to an empty array created with
    # `np.empty(correct_shape, dtype=object)`.
    numpy.array(batch_output, dtype=object)
```
